feat(cfw): add resource CFW capture (#5238)
huaweicloud/terraform-provider-huaweicloud: docs/resources/cfw_capture.md

---
subcategory: "Cloud Firewall (CFW)"
layout: "huaweicloud"
page_title: "HuaweiCloud: huaweicloud_cfw_capture"

> description: ""
> ---

Reviewer: ```suggestion
description: -|
  Manages a CFW capture resource within HuaweiCloud.
---
```

# huaweicloud_cfw_capture

Manages a CFW capture resource within HuaweiCloud.

-> **NOTE:** For the Cloud Firewall service, you can only initiate up to 20 packet capture tasks per day.
Beyond this limit, no additional packet captures can be initiated. Furthermore, only one packet capture task can be in
progress at any given time.

## Example Usage

```hcl
variable "fw_instance_id" {}
variable "name" {}
variable "duration" {}
variable "max_packets" {}

resource huaweicloud_cfw_capture "test" {
  fw_instance_id = var.fw_instance_id
  name           = var.name
  duration       = var.duration
  max_packets    = var.max_packets
  

>   destination {
>     address = "1.1.1.1"
>     address_type = 0
>   }

Reviewer: ```suggestion
  destination {
    address      = "1.1.1.1"
    address_type = 0
  }
```

>   source {
>     address = "2.2.2.2"
>     address_type = 0
>   }

Reviewer: ```suggestion
  source {
    address      = "2.2.2.2"
    address_type = 0
  }
```

* `source` - (Required, List) The source configuration.
  The [source](#Source) structure is documented below.

* `stop_capture` - (Optional, Bool) Whether to stop the capture.

<a name="Destination"></a>
The `destination` block supports:

* `address` - (Required, String) The address.

* `address_type` - (Required, Int) The address type.

<a name="Service"></a>
The `service` block supports:

* `protocol` - (Required, Int) The protocol type.

* `dest_port` - (Optional, String) The destination port.

* `source_port` - (Optional, String) The source port.

<a name="Source"></a>
The `source` block supports:

* `address` - (Required, String) The address.

* `address_type` - (Required, Int) The address type.

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

* `id` - The resource ID.

* `created_at` - The creation time of the capture task.

* `status` - The status of the capture task.

* `updated_at` - The update time of the capture task.

## Timeouts

This resource provides the following timeouts configuration options:

* `update` - Default is 5 minutes.
* `delete` - Default is 5 minutes.

## Import

> The capture can be imported using `fw_instance_id`, `name`, separated by slashes, e.g.

Reviewer: ```suggestion
The capture can be imported using `fw_instance_id`, `name`, separated by  a slashe (/), e.g.
```

---

# huaweicloud_cfw_capture

Manages a CFW capture resource within HuaweiCloud.

-> **NOTE:** For the Cloud Firewall service, you can only initiate up to 20 packet capture tasks per day.
Beyond this limit, no additional packet captures can be initiated. Furthermore, only one packet capture task can be in
progress at any given time.

## Example Usage

```hcl
variable "fw_instance_id" {}
variable "name" {}
variable "duration" {}
variable "max_packets" {}

resource huaweicloud_cfw_capture "test" {
  fw_instance_id = var.fw_instance_id
  name           = var.name
  duration       = var.duration
  max_packets    = var.max_packets
  
  destination {
    address      = "1.1.1.1"
    address_type = 0
  }

  source {
    address      = "2.2.2.2"
    address_type = 0
  }

  service {
    protocol = -1
  }
}
```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
  If omitted, the provider-level region will be used.
  Changing this creates a new resource.

* `fw_instance_id` - (Required, String, NonUpdatable) Specifies the ID of the firewall instance.

> * `name` - (Required, String, NonUpdatable) Specifies the capture name.

Reviewer 2: ```suggestion
* `name` - (Required, String, NonUpdatable) Specifies the capture task name.
```
and fix other places

* `max_packets` - (Required, Int, NonUpdatable) Specifies the maximum number of packets captured.
  The Maximum value is **1000000**.

* `destination` - (Required, List, NonUpdatable) Specifies the destination configuration.
  The [destination](#Destination) structure is documented below.

* `source` - (Required, List, NonUpdatable) Specifies the source configuration.
  The [source](#Source) structure is documented below.

* `service` - (Required, List, NonUpdatable) Specifies the service configuration.
  The [service](#Service) structure is documented below.

* `stop_capture` - (Optional, Bool) Specifies whether to stop the capture.

<a name="Destination"></a>
The `destination` block supports:

* `address` - (Required, String, NonUpdatable) Specifies the address.

* `address_type` - (Required, Int, NonUpdatable) Specifies the address type.
  The valid values are:
  + **0**: indicates IPv4;
  + **1**: indicates IPv6.

<a name="Source"></a>
The `source` block supports:

* `address` - (Required, String, NonUpdatable) Specifies the address.

* `address_type` - (Required, Int, NonUpdatable) Specifies the address type.
  The valid values are:
  + **0**: indicates IPv4;

>   + **1**: indicates IPv6.

Reviewer 3: Destination and Source blocks are same, maybe you can name it as `addresses` block.

  + **17**: indicates UDP;
  + **1**: indicates ICMP;
  + **58**: indicates ICMPv6;
  + **-1**: indicates any protocol.

* `dest_port` - (Optional, String, NonUpdatable) Specifies the destination port.

* `source_port` - (Optional, String, NonUpdatable) Specifies the source port.

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

* `id` - The resource ID.

* `created_at` - The creation time of the capture task.

* `status` - The status of the capture task.

* `updated_at` - The update time of the capture task.

## Timeouts

This resource provides the following timeouts configuration options:

* `update` - Default is 5 minutes.
* `delete` - Default is 5 minutes.

## Import

The capture can be imported using `fw_instance_id`, `name`, separated by a slash, e.g.

```bash
$ terraform import huaweicloud_cfw_capture.test <fw_instance_id>/<name>
```

Note that the imported state may not be identical to your resource definition, due to some attributes missing from the
API response, security or some other reason.
The missing attributes is `stop_capture`. It is generally recommended running `terraform plan` after importing the resource.

> You can then decide if changes should be applied to the instance, or the resource definition should be updated to

Reviewer 2: ```suggestion
You can then decide if changes should be applied to the  capture, or the resource definition should be updated to
```
